src/components/AddTask.jsx

```diff
@@ -0,0 +1,72 @@
+import { ReactComponent as DeleteIcon } from 'assets/delete.svg';
+import Button from 'components/ui/Button';
+import { useEffect, useRef, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { addTodo, setIsAddingTask } from 'redux/actions/TodoAction';
+import { KEY_ENTER, RESPONSE_ERROR } from 'utils/constants';
+import { validate } from 'utils/helpers/index';
+
+function AddTask() {
+  const [error, setError] = useState(null);
+  const [taskDetails, setTaskDetails] = useState('');
+  const dispatch = useDispatch();
+  const textAreaRef = useRef(null);
+
+  function onSubmit(e) {
+    e.preventDefault();
+    const validateDetails = validate(taskDetails);
+
+    if (validateDetails.status === RESPONSE_ERROR) {
+      setError(validateDetails.message);
+      return;
+    }
+
+    setError(null);
+    dispatch(addTodo(validateDetails.text));
+    dispatch(setIsAddingTask(false));
+    setTaskDetails('');
+  }
+
+  function onTyping(e) {
+    setTaskDetails(e.target.value);
+  }
+
+  function onCancel(e) {
+    e.preventDefault();
+    dispatch(setIsAddingTask(false));
+  }
+
+  function onKeyDown(e) {
+    if (e.key === KEY_ENTER) {
+      onSubmit(e);
+    }
+  }
```

Comment:
Bhai, as we know that Arrow functions can provide a more concise syntax and can make the code easier to read and understand, can we convert them into Arrow functions?